prover/src/domain.rs

```diff
@@ -23,6 +23,12 @@ pub struct StarkDomain<B: StarkField> {
 
     /// Offset of the low-degree extension domain.
     domain_offset: B,
+
+    /// [g^i for i in (0..ce_domain_size)] where g is the constraint evaluation domain generator.
+    pub domain_g: Vec<B>,
```

Comment:
nit: I'd name this `ce_domain` to be consistent with the naming convention. Also, we can get rid of `ce_domain_size` field as we can get it simply as `self.ce_domain.len()`.

Also, I wouldn't make this public and would instead add accessor functions.